packages/zipkin/test/httpServerInstrumentation.test.js

```diff
@@ -107,6 +107,55 @@ describe('Http Server Instrumentation', () => {
     expect(annotations[8].annotation.annotationType).to.equal('ServerSend');
   });
 
+  it('should receive flags from the client', () => {
+      const record = sinon.spy();
+      const recorder = {record};
+      const ctxImpl = new ExplicitContext();
+      const tracer = new Tracer({recorder, ctxImpl});
+
+      const headers = {
+          'X-B3-Flags': '1'
+      };
+      const port = 80;
+      const url = `http://127.0.0.1:${port}`;
+      const instrumentation = new HttpServer({tracer, serviceName: 'service-a', port});
+      const readHeader = function(name) { return headers[name] ? new Some(headers[name]) : None; };
+      ctxImpl.scoped(() => {
+          const id = instrumentation.recordRequest('POST', url, readHeader);
+          tracer.recordBinary('message', 'hello from within app');
+          instrumentation.recordResponse(id, 202);
+      });
+      const annotations = record.args.map(args => args[0]);
+
+      expect(annotations[0].annotation.annotationType).to.equal('ServiceName');
+      expect(annotations[0].annotation.serviceName).to.equal('service-a');
+
+      expect(annotations[1].annotation.annotationType).to.equal('Rpc');
+      expect(annotations[1].annotation.name).to.equal('POST');
+
+      expect(annotations[2].annotation.annotationType).to.equal('BinaryAnnotation');
+      expect(annotations[2].annotation.key).to.equal('http.url');
+      expect(annotations[2].annotation.value).to.equal(url);
+
+      expect(annotations[3].annotation.annotationType).to.equal('ServerRecv');
+
+      expect(annotations[4].annotation.annotationType).to.equal('LocalAddr');
+
+      expect(annotations[5].annotation.annotationType).to.equal('BinaryAnnotation');
+      expect(annotations[5].annotation.key).to.equal('X-B3-Flags');
```

Comment:
just noticing.. this is weird... do we literally capture headers as tags (aka binary annotations?)

Reply:
It is, I could remove it but since this is a copy fromt he other test I would prefer to keep it an remove in a next PR.